### PR TITLE
build(ui): remove no longer needed `--openssl-legacy-provider`

### PR DESCRIPTION
**Pull request by agilgur5**
### Motivation

- this was needed for legacy Webpack v4 support, but as of #11628, we are on Webpack v5 and so this is no longer necessary
  - and this is more secure too, as the legacy option was insecure
  
Remembered that this was no longer needed after https://github.com/argoproj/argo-ui/pull/432#issuecomment-1879816619, I must've forgot I could remove this earlier (or that the flags were even there)

### Modifications

- basically undoes the parts of #11410 that added `--openssl-legacy-provider`
  - removes all `--openssl-legacy-provider` flags

### Verification

1. `yarn build` still works fine
1. `yarn start` still works fine

### CODE REVIEW

**Comment by alexec** [comment]
I’m trying to remember why this was added, but I think if the build passes, this is good to go.

**Comment by agilgur5** [comment]
I wrote why it was added in my PR description

**Review by jmeridth — APPROVED** [review]

**Review by terrytangyuan — APPROVED** [review]
